project/src/test/java/com/google/sps/SearchDataServletTest.java

```diff
@@ -0,0 +1,137 @@
+  private HttpServletRequest postRequest;
+  private HttpServletResponse postResponse;
+  private HttpServletRequest secondPostRequest;
+  private HttpServletResponse secondPostResponse;
+  private HttpServletRequest getRequest;
+  private HttpServletResponse getResponse;
+  private StringWriter stringWriter;
+  private PrintWriter printWriter;
+  private SearchDataServlet searchDataServlet;
+  private static final String EVENT_ID_1 = TestUtils.newRandomId();
+  private static final String EVENT_ID_2 = TestUtils.newRandomId();
+  private static final String PARAMETER_KEYWORD = "keyword";
+  private static final String PARAMETER_EVENT_ID = "event-id";
+  private static final String PARAMETER_DESCRIPTION = "description";
+  private static final String WALKING = "walking";
+  private static final String CAKE = "cake";
+  private static final String MUSIC = "music";
+  private static final String DESCRIPTION_WITH_MUSIC_KEYWORD =
+      "This is an annual event held at the State Capitol to celebrate agricultures."
+          + "The event will include food, live music and educational booths with public speakers."
+          + "Event hours is 9 am- 1 pm. 2550 attendees anticipated.";
+  private static final String DESCRIPTION_WITH_WALKING_KEYWORD =
+    "Sutter Middle School will be walking to McKinley Park. 7th grade class and teachers"
+    + " will have a picnic, play games and eat lunch at the park and Clunie Pool.";
+  private static final String DESCRIPTION_WITHOUT_CAKE_KEYWORD =
+    "Sutter Middle School will be walking to McKinley Park. 7th grade class and teachers"
+    + " will have a picnic, play games and eat lunch at the park and Clunie Pool.";
+
+  @Before
+  public void setUp() throws Exception {
+    postRequest = Mockito.mock(HttpServletRequest.class);
+    postResponse = Mockito.mock(HttpServletResponse.class);
+
+    secondPostRequest = Mockito.mock(HttpServletRequest.class);
+    secondPostResponse = Mockito.mock(HttpServletResponse.class);
+
+    getRequest = Mockito.mock(HttpServletRequest.class);
+    getResponse = Mockito.mock(HttpServletResponse.class);
+
+    stringWriter = new StringWriter();
+    printWriter = new PrintWriter(stringWriter);
+    Mockito.when(getResponse.getWriter()).thenReturn(printWriter);
+
+    searchDataServlet = new SearchDataServlet();
+  }
+
+  @Test
+  public void addEventAndRetrieveResultsForKeywordNotInEventDescription_noResultsReturned()
+      throws IOException {
+    Mockito.when(postRequest.getParameter(PARAMETER_DESCRIPTION))
+        .thenReturn(DESCRIPTION_WITHOUT_CAKE_KEYWORD);
+    Mockito.when(postRequest.getParameter(PARAMETER_EVENT_ID)).thenReturn(EVENT_ID_1);
+    Mockito.when(getRequest.getParameter(PARAMETER_KEYWORD)).thenReturn(CAKE);
+
+    searchDataServlet.doPost(postRequest, postResponse);
+    searchDataServlet.doGet(getRequest, getResponse);
+
+    Assert.assertEquals(CommonUtils.convertToJson(Arrays.asList()), stringWriter.toString().trim());
+  }
+
+  @Test
```

Comment:
I think you can remove this test. The 2 results test below should be already enough if you don't have any exclusive logic for one result.

Reply:
Done - I see.